sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -481,6 +530,53 @@ def create_job(
                 f"{len(volumes or [])} vs. {len(volume_mounts or [])}"
             )
 
+        # If labels were provided, validate Kueue queue if specified
+        if local_queue is not None and labels is not None:
+            # Ensure metadata exists
+            if not hasattr(job, "metadata") or job.metadata is None:
+                job.metadata = models.V1ObjectMeta(
+                    name=getattr(job.metadata, "name", name),
+                    namespace=getattr(job.metadata, "namespace", namespace),
+                )
+            # Validate Kueue queue existence if label key is present
+            queue_label = "kueue.x-k8s.io/queue-name"
+            if queue_label in labels:
+                queue_name = labels[queue_label]
+                if queue_name != local_queue:
+                    logger.warning(
+                        f"Provided local_queue '{local_queue}' does not match "
```

Comment:
I would fail in that case? Or maybe the local queue and the label should be mutually exclusive.